src/main/java/com/blackducksoftware/integration/hub/builder/HubServerConfigBuilder.java

```diff
@@ -62,11 +68,46 @@
 
     private String ignoredProxyHosts;
 
+    private boolean autoImportHttpsCertificates;
+
+    private String keystorePassword;
+
+    private IntLogger logger;
+
     public HubServerConfigBuilder() {
         timeoutSeconds = String.valueOf(DEFAULT_TIMEOUT_SECONDS);
     }
 
     @Override
+    public HubServerConfig build() throws IllegalStateException {
+        try {
+            return super.build();
+        } catch (final IllegalStateException e) {
+            if (e.getMessage().contains("SunCertPathBuilderException")) {
+                if (autoImportHttpsCertificates) {
+                    final HubCertificateHandler handler = new HubCertificateHandler(getLogger());
+                    try {
+                        if (StringUtils.isNotBlank(hubUrl)) {
```

Comment:
Is it possible for the hub url to be empty at this point?  if we get the SunCertPathBuilderException we tried to connect to a hub url correct?